packages/now-cli/src/util/dev/builder.ts

```diff
@@ -224,6 +224,19 @@ export async function executeBuild(
           ? buildResultOrOutputs.distPath
           : undefined,
     };
+  } else if (builder.version === 3) {
+    const { output, ...rest } = buildResultOrOutputs;
+
+    if ((output as BuilderOutput).type !== 'Lambda') {
```

Comment:
I think you can separate the possible output types by doing:

```
if (Array.isArray(output)) {
  throw new Error(`The result of "builder.build" must be a single Lambda, not an array'`);
}

if (output.type !== 'Lambda') {
  throw new Error(`The result of "builder.build" must be of type Lambda'`);
}
```

Reply:
Unfortunately that doesn't work, I couldn't find a way around the casting

Reply:
@styfle I've tried to reduce the casting. However, those are sort of assertion checks, so if we wouldn't cast them TypeScript would complain that this is always `true`, but this is the builder result, which could actually be `any`, since we don't control what it returns.